flink-optimizer/src/test/java/org/apache/flink/optimizer/operators/CoGroupGlobalPropertiesCompatibilityTest.java

```diff
@@ -95,6 +97,32 @@ public int partition(Object key, int numPartitions) {
 				
 				assertTrue(descr.areCompatible(reqLeft, reqRight, propsLeft, propsRight));
 			}
+
+			TestDistribution dist1 = new TestDistribution(1);
+			TestDistribution dist2 = new TestDistribution(1);
+			
+			// test compatible range partitioning
+			{
```

Comment:
Can you add a checks with two keys, one DESC and one ASC? 
Orders of both should be the same to check if it is correctly identified as compatible.